### DISCUSSION TITLE
v8.0.0

### DISCUSSION THREAD

**Discussion by tadashi-aikawa** [tadashi-aikawa/obsidian-various-complements-plugin#183] [Announcements]
### Fuzzy match (#127) `🏷8.0.0-beta10`

I have added a "Fuzzy match" option which enables you to match fuzzy. There should be little or no performance degradation, but it may take twice the time, depending on the case.

https://user-images.githubusercontent.com/9500018/219951893-7e5cddee-5f69-4546-b4e9-19a181406079.mp4

----

> **Note**
> **The default value of "Fuzzy match" is `true`. If you don't prefer it, please set `false` to it. `🏷8.0.0-beta12`**

### "Insert an alias that is transformed from the displayed internal link" option ` (#193 )🏷8.0.0-beta11`

As the Vault grows, there are notes with the same name in different contexts. Wikipedia has addressed these issues by adding bracketed suffixes to note names and removing the suffixes from the link display string. However, it also means we must insert the internal link with the alias. It is not efficient.

For these reasons, I have added an "Insert an alias that is transformed from the displayed internal link" option. For example, if you set 

- `Before: regular expression pattern with captures`: `(?<name>.+) \(.+\)$`
- `After`: `$<name>`

Then the plugin inserts the link, `[[hoge (huga)]]`, as `[[hoge (huga)|hoge]]`.

https://user-images.githubusercontent.com/9500018/220118611-a42f316e-d201-4901-a27e-5af581ed1b83.mp4

## ✨ **New things**

- [x] Add a "Patterns to suppress trigger" option (#162) `🏷8.0.0-beta2`
- [x] Add an "Update internal links on save" option `🏷8.0.0-beta4` `🏷8.0.0-beta5`

## 🔼 **Improvements**

- [x] In internal link complement, when multiple files with different paths but the same name exist, insert links that display only the file name using aliases (#193) `🏷8.0.0-beta3`
- [x] Add a "CC-CEDICT path" option when Strategy is chinese (#169) (#178) `🏷8.0.0-beta1`

![image](https://user-images.githubusercontent.com/9500018/204120761-f7cf931c-8df1-4d0c-ade8-654381ed3e2a.png)

---

## Beta releases

- [x] 🚀 [v8.0.0-beta12](https://github.com/tadashi-aikawa/obsidian-various-complements-plugin/releases/tag/8.0.0-beta12)
- [x] 🚀 [v8.0.0-beta11](https://github.com/tadashi-aikawa/obsidian-various-complements-plugin/releases/tag/8.0.0-beta11)
- [x] 🚀 [v8.0.0-beta10](https://github.com/tadashi-aikawa/obsidian-various-complements-plugin/releases/tag/8.0.0-beta10)
- [x] 🚀 [v8.0.0-beta9](https://github.com/tadashi-aikawa/obsidian-various-complements-plugin/releases/tag/8.0.0-beta9)
- [x] 🚀 [v8.0.0-beta8](https://github.com/tadashi-aikawa/obsidian-various-complements-plugin/releases/tag/8.0.0-beta8)
- [x] 🚀 [v8.0.0-beta7](https://github.com/tadashi-aikawa/obsidian-various-complements-plugin/releases/tag/8.0.0-beta7)
- [x] 🚀 [v8.0.0-beta6](https://github.com/tadashi-aikawa/obsidian-various-complements-plugin/releases/tag/8.0.0-beta6)
- [x] 🚀 [v8.0.0-beta5](https://github.com/tadashi-aikawa/obsidian-various-complements-plugin/releases/tag/8.0.0-beta5)
- [x] 🚀 [v8.0.0-beta4](https://github.com/tadashi-aikawa/obsidian-various-complements-plugin/releases/tag/8.0.0-beta4)
- [x] 🚀 [v8.0.0-beta3](https://github.com/tadashi-aikawa/obsidian-various-complements-plugin/releases/tag/8.0.0-beta3)
- [x] 🚀 [v8.0.0-beta2](https://github.com/tadashi-aikawa/obsidian-various-complements-plugin/releases/tag/8.0.0-beta2)
- [x] 🚀 [v8.0.0-beta1](https://github.com/tadashi-aikawa/obsidian-various-complements-plugin/releases/tag/8.0.0-beta1)

**Comment by tadashi-aikawa** [comment]
>  Add an "Update internal links on save" option 🏷8.0.0-beta4

https://user-images.githubusercontent.com/9500018/218266116-94daae86-6ed7-4744-9844-7f6f266e53db.mp4

I have optimized performance.

- The logic to decide whether it needs to reindex internal links consumes only 0.000005s
- It reindexes internal links only if there are differences in unresolved links in the active file.

![image](https://user-images.githubusercontent.com/9500018/218266396-7d7ff3f5-ae6a-4678-af66-78d5d212afe4.png)

**Comment by Glint-Eye** [comment]
The fuzzy matching is a great feature. I'd have a suggestion to maybe highlight the matching characters with a bold character (like obsidian quick switcher does). For example in the video example for fuzzy matching with "dictionary"

dina
**di**ctio**na**ry

cheers :)

**Reply by tadashi-aikawa** [reply]
Thank you for your suggestion👍 

Could you create a new issue?

※ But I will not implement it soon because there are many specifications to consider, and some refactoring is needed. I appreciate your understanding in advance.
